### PR TITLE
chore: Add bump_version workflow

### PR DESCRIPTION
**Pull request by zyannes**
### Security
- [ ] Changelog has been updated
- [ ] Change has no security implications (otherwise, ping the security team)

### CODE REVIEW

File: .github/workflows/bump_version.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        run: |
+          # check if the branch already has a pull request open
+          if gh pr list --head ${SOURCE} | grep -vq "no pull requests"; then
+              # pull request already open
+              echo "pull request from SOURCE ${SOURCE} to TARGET ${TARGET} is already open";
+              echo "cancelling release"
+              exit 1
+          fi
+          # open new pull request with the body of from the local template.
+          res=$(gh pr create --title "${TITLE}" --body "Bump Semgrep Version to ${NEW_SEMGREP_VERSION}" \
```
Comment (spencerdrak) [inline .github/workflows/bump_version.yml]: can we also just remove the output capture here? makes it easier to debug if anything goes wrong.

**Review by spencerdrak — APPROVED** [review]
LGTM - will the `semgrep-ci` bot need additional perms to be able to run this? 

I suspect yes, and I'll work with chris to adjust that.

File: .github/workflows/bump_version.yml (hunk 1 above)
Comment (zyannes) [reply .github/workflows/bump_version.yml]: Good call! Updated in latest commit

**Comment by zyannes** [comment]
@spencerdrak The `semgrep-ci` bot will need to have `checks: write` and `pull_requests: write` permissions. I suspect it already has these permissions since this workflow performs similar API calls as the `release-action` workflow. I'll do a simple test once this PR gets merged in to check if the existing permissions are sufficient.

**Pull request merged**